packages/investor-foxy/src/api/api.ts

```diff
@@ -668,7 +670,19 @@ export class FoxyApi {
       (pastTokeCycleIndex && stakingTokenAvailableWithTokemak.gte(coolDownInfo.amount)) ||
       stakingTokenAvailable
 
-    return epochExpired && coolDownValid && validCycleAndAmount
+    const epochsLeft = bnOrZero(coolDownInfo.endEpoch.toString()).minus(epoch.number.toString())
+    const blocksLeftInCurrentEpoch =
+      epochsLeft.gt(0) && epoch.endBlock.gt(currentBlock)
+        ? epoch.endBlock.sub(currentBlock).toNumber()
+        : 0 // calculate time remaining in current epoch
+    const blocksLeftInFutureEpochs = epochsLeft.minus(1).gt(0)
+      ? epochsLeft.minus(1).times(epoch.length).toNumber()
+      : 0
+
+    return (
+      (!blocksLeftInCurrentEpoch && !blocksLeftInFutureEpochs) || // satisfying getTimeUntilClaimable constraints
```

Comment:
could maybe add the one extra var and equals check to make it a little more readable imo, but works either way
```
    const blocksUntilClaimable = bnOrZero(blocksLeftInCurrentEpoch).plus(blocksLeftInFutureEpochs) // total blocks left until can claim
   blocksUntilClaimable.eq(0) || ...
```

Reply:
Absolutely, I can't say I understand much of it either way, FOXy contract / investor-foxy is a black box to me, the goal was really to be a 1/1 between the two methods. Will definitely do a bit more cleanup there when bringing it to web!